internal/cmd/cli/target.go

```diff
@@ -0,0 +1,189 @@
+package cli
+
+import (
+	"context"
+	"flag"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	command "github.com/rancher/fleet/internal/cmd"
+	"github.com/rancher/fleet/internal/cmd/controller/target"
+	"github.com/rancher/fleet/internal/content"
+	"github.com/rancher/fleet/internal/manifest"
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	zopts  = zap.Options{Development: true}
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(v1alpha1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+// NewTarget returns a subcommand to print available targets for a bundle
+func NewTarget() *cobra.Command {
+	cmd := command.Command(&Target{}, cobra.Command{
+		Short: "Print available targets for a bundle",
+	})
+	cmd.SetOut(os.Stdout)
+	// add command line flags from zap and controller-runtime, which use
+	// goflags and convert them to pflags
+	fs := flag.NewFlagSet("", flag.ExitOnError)
+	zopts.BindFlags(fs)
+	ctrl.RegisterFlags(fs)
+	cmd.Flags().AddGoFlagSet(fs)
+	return cmd
+}
+
+type Target struct {
+	BundleFile string `usage:"Location of the Bundle resource yaml" short:"b"`
+	ListInputs bool   `usage:"List the live resources, which are used in targeting, like clusters, as YAML" short:"l"`
+
+	Namespace string `usage:"Override the namespace of the bundle. Targeting searches this namespace for clusters." short:"n"`
+}
+
+func (t *Target) Run(cmd *cobra.Command, args []string) error {
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&zopts)))
+	ctx := log.IntoContext(cmd.Context(), ctrl.Log)
+
+	if t.BundleFile == "" {
+		return cmd.Help()
+	}
+
+	b, err := os.ReadFile(t.BundleFile)
+	if err != nil {
+		return err
+	}
+	bundle := &v1alpha1.Bundle{}
+	err = yaml.Unmarshal(b, bundle)
+	if err != nil {
+		return err
+	}
+	// needs to be set, fleet will only create this many deployments if the bundle is new
+	bundle.Status.MaxNew = 50
```

Comment:
I don't want to fix this.. let's revisit it when we deal with the rollout strategy. We cab just set it to one, since the CLI will only generate one bundledeployment?

```suggestion
	bundle.Status.MaxNew = 1
```

Reply:
Works for me :)

Reply:
Sorry, this got "resolved" when I added my code suggestion to the batch